com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs

```diff
@@ -97,16 +97,27 @@ bool StripUnusedFeatures(ShaderFeatures features, Shader shader, ShaderSnippetDa
             if (!IsFeatureEnabled(features, ShaderFeatures.AdditionalLightShadows) && isAdditionalLightShadow)
                 return true;
 
-            // Additional light are shaded per-vertex. Strip additional lights per-pixel and shadow variants
+
+            // Additional light are shaded per-vertex or per-pixel.
             bool isAdditionalLightPerPixel = compilerData.shaderKeywordSet.IsEnabled(m_AdditionalLightsPixel);
-            if (IsFeatureEnabled(features, ShaderFeatures.VertexLighting) &&
-                (isAdditionalLightPerPixel || isAdditionalLightShadow))
-                return true;
+            bool isAdditionalLightPerVertex = compilerData.shaderKeywordSet.IsEnabled(m_AdditionalLightsVertex);
+            if (isAdditionalLightPerPixel || isAdditionalLightShadow)
+            {
+                bool isPerPixelLightingFeatureEnabled = IsFeatureEnabled(features, ShaderFeatures.AdditionalLights);
+                bool isPerVertexLightingFeatureEnabled = IsFeatureEnabled(features, ShaderFeatures.VertexLighting);
 
-            // No additional lights
-            if (!IsFeatureEnabled(features, ShaderFeatures.AdditionalLights) &&
-                (isAdditionalLightPerPixel || isAdditionalLightShadow || compilerData.shaderKeywordSet.IsEnabled(m_AdditionalLightsVertex)))
-                return true;
+                // No additional lights
+                if (!isPerPixelLightingFeatureEnabled && !isPerVertexLightingFeatureEnabled)
+                    return true;
+
+                // Only Per-pixel is selected in every URP asset
+                if (!isPerVertexLightingFeatureEnabled && isAdditionalLightPerVertex)
```

Comment:
are we sure this conditional is ever hit? It seems isAdditionalLightsPerVertex should always be false here, since we are checking it inside a if (isAdditionalLightsPerPixel) and these keywords are mutually exclusive. (multi_compile _ADDITIONAL_LIGHTS_VERTEX _ADDITIONAL_LIGHTS)

Reply:
You're right. I believed I've now fixed that check.